### ISSUE TITLE
TextBox ContextMeny crashes app when a MenuItem Style is applied

### ISSUE THREAD

**Issue by cornerbowlsoftware** [AvaloniaUI/Avalonia#16953]
### Describe the bug

I have the following style defined so I can bind the MenuItem properties to my MenuItemViewModels:

```
	<Style Selector="MenuItem" x:DataType="lib:MenuItemViewModel">
		<Setter Property="Header" Value="{Binding Text}"/>
		<Setter Property="Command" Value="{Binding Command}"/>
		<Setter Property="InputGesture" Value="{Binding InputGestureText, Converter={StaticResource StringToKeyGestureConverter}}"/>
		<Setter Property="Icon" Value="{Binding PathIcon, Converter={StaticResource PathToIconConverter}}"/>
		<Setter Property="ItemsSource" Value="{Binding Items}" />
	</Style>
```

Once defined, any right click over any TextBox causes the application to crash with the following error:

`Unable to cast object of type 'MyCurrentViewModel' to type 'MenuItemViewModel'.`

### To Reproduce

Create an app with a TextBox, define the style, run the app, right click on the TextBox.

### Expected behavior

The TextBox control should display its' own ContextMenu without trying to apply my style or my bindings.

### Avalonia version

11.2.0-beta-1

**Comment by timunie** [comment]
Your Data context is wrong. x:DataType is only a hint for the compiler, it won't create a new instance or anything

**Comment by cornerbowlsoftware** [comment]
@timunie Yes I wouldn't expect it to create an instance.   The built-in TextBox is trying to use my MenuItem style hence the incorrect DataContext.

**Comment by timunie** [comment]
How should it know which to use? You need to move the style somewhere else in that case where it's applicable

**Comment by cornerbowlsoftware** [comment]
@timunie Ok that makes sense, however I never applied a style to Menu Items that sit inside a Context Menu.  Is there a way for me to specify the Menu Item style for my ContextMenu?

`<ContextMenu ItemsSource="{Binding MenuItemVMs}"/>`

**Comment by timunie** [comment]
ContextMenu.Styles or ItemContainerTheme if you want to digg into that. Its similar to styles but you can define it once and consume it via StaticResource.

**Comment by cornerbowlsoftware** [comment]
@timunie.  Ok thanks.
